# 1 Table 1


## Question 1.


In [4]:
import pandas as pd
import statsmodels.api as sm
import statsmodels.formula.api as smf

df = pd.read_csv("charitable_giving.csv")

model = smf.ols(formula = 'months_since_last_donation ~ treatment',
               data = df).fit()
print(model.summary())

                                OLS Regression Results                                
Dep. Variable:     months_since_last_donation   R-squared:                       0.000
Model:                                    OLS   Adj. R-squared:                 -0.000
Method:                         Least Squares   F-statistic:                   0.01428
Date:                        Thu, 25 Jan 2024   Prob (F-statistic):              0.905
Time:                                21:47:52   Log-Likelihood:            -1.9585e+05
No. Observations:                       50082   AIC:                         3.917e+05
Df Residuals:                           50080   BIC:                         3.917e+05
Df Model:                                   1                                         
Covariance Type:                    nonrobust                                         
                 coef    std err          t      P>|t|      [0.025      0.975]
---------------------------------------------------

The average “Number of months since last donation” is not significantly different between treatment and control, indicated by the high p-value at 0.905. 

The intercept 12.9981 indicates the response rate in the control group, on average.

The treatment regression coefficients, 0.0137, indicates the variation in the response rate between the treatment group and the control group.

The regression-based means consider the association between the treatment variable and the dependent variable, offering coefficient estimates that reveal the direction and magnitude of the treatment's impact.

The group-specific means reported in Table 1 of the paper offers a straightforward comparison of the means between the treatment and control groups.

## Question 2.

The difference in “Number of month since last donation” between treatment and control is not statistically significant at the 95% confidence level, because the p-value is larger than 0.05. 

This is the result I expected because treatment and control groups are supposed to be alike.

## Question 3.

The take-away from Table 1 in the paper is that the allocation to treatment and control groups was independent of observable demographic information and previous donation history, which indicates randomization.

# 2 Response rate regressions

## Question 1.

In [7]:
import pandas as pd
import statsmodels.api as sm
import statsmodels.formula.api as smf

df = pd.read_csv("charitable_giving.csv")

model = smf.ols(formula = 'donation_dummy ~ treatment',
               data = df).fit()
print(model.summary())

                            OLS Regression Results                            
Dep. Variable:         donation_dummy   R-squared:                       0.000
Model:                            OLS   Adj. R-squared:                  0.000
Method:                 Least Squares   F-statistic:                     9.618
Date:                Thu, 25 Jan 2024   Prob (F-statistic):            0.00193
Time:                        21:59:40   Log-Likelihood:                 26630.
No. Observations:               50083   AIC:                        -5.326e+04
Df Residuals:                   50081   BIC:                        -5.324e+04
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
Intercept      0.0179      0.001     16.225      0.0

The intercept coefficient, 0.0179, indicates the response rate in the control group, on average.

The treatment coefficient, 0.0042, indicates the variation in the response rate between the treatment group and the control group.

Compare to the results presented in the first row of Table 2a, 0.018 and 0.022, each is the intercept, the average response rates in the treatment and control groups.

## Question 2.

In [9]:
import pandas as pd
import statsmodels.api as sm
import statsmodels.formula.api as smf

df = pd.read_csv("charitable_giving.csv")

model = smf.ols(formula = 'donation_dummy ~ ratio1 + ratio2 + ratio3',
               data = df).fit()
print(model.summary())

                            OLS Regression Results                            
Dep. Variable:         donation_dummy   R-squared:                       0.000
Model:                            OLS   Adj. R-squared:                  0.000
Method:                 Least Squares   F-statistic:                     3.665
Date:                Thu, 25 Jan 2024   Prob (F-statistic):             0.0118
Time:                        22:17:08   Log-Likelihood:                 26630.
No. Observations:               50083   AIC:                        -5.325e+04
Df Residuals:                   50079   BIC:                        -5.322e+04
Df Model:                           3                                         
Covariance Type:            nonrobust                                         
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
Intercept      0.0179      0.001     16.225      0.0

The intercept coefficient, 0.0179, indicates the response rate in the control group, on average.

The ratio1 coefficient, 0.0029, indicates the variation in the response rate between the ratio1 group and the control group.

The ratio2 coefficient, 0.0048, indicates the variation in the response rate between the ratio2 group and the control group.

The ratio3 coefficient, 0.0049, indicates the variation in the response rate between the ratio3 group and the control group.

## Question 3.

In [11]:
import pandas as pd
import statsmodels.api as sm
import statsmodels.formula.api as smf

df = pd.read_csv("charitable_giving.csv")

model = smf.ols(formula = 'donation_dummy ~ ratio1 + ratio2 + ratio3',
               data = df).fit()

ratio1_coef = model.params['ratio1']
ratio2_coef = model.params['ratio2']
diff = ratio2_coef - ratio1_coef
diff

0.0018842510217149415

The difference in impacts on response rate between the 1:1 and 2:1 match ratios is 0.0018842510217149415.

## Question 4.

Based on the regressions I ran and more generally the results in Table 2a, I conclude that using the 2:1 match ratio is not much more effective than 1:1, although matching increases donations.

# 3 Response rates in red/blue states

## Question 1.

In [21]:
import pandas as pd
import statsmodels.api as sm
import statsmodels.formula.api as smf

df = pd.read_csv("charitable_giving.csv")
df_red = df[df['red_state_dummy'] == 1]

model_red = smf.ols(formula = 'donation_dummy ~ treatment',
               data = df_red).fit()
print(model_red.summary())

                            OLS Regression Results                            
Dep. Variable:         donation_dummy   R-squared:                       0.001
Model:                            OLS   Adj. R-squared:                  0.001
Method:                 Least Squares   F-statistic:                     17.24
Date:                Thu, 25 Jan 2024   Prob (F-statistic):           3.31e-05
Time:                        22:40:26   Log-Likelihood:                 10839.
No. Observations:               20242   AIC:                        -2.167e+04
Df Residuals:                   20240   BIC:                        -2.166e+04
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
Intercept      0.0146      0.002      8.398      0.0

The intercept coefficient, 0.0146, indicates the response rate in the control group, on average, in red states.

The treatment coefficient, 0.0088, indicates the variation in the response rate between the treatment group and the control group in red states.

In [26]:
import pandas as pd
import statsmodels.api as sm
import statsmodels.formula.api as smf

df = pd.read_csv("charitable_giving.csv")
df_blue = df[df['red_state_dummy'] == 0]

model_blue = smf.ols(formula = 'donation_dummy ~ treatment',
               data = df_blue).fit()
print(model_blue.summary())

                            OLS Regression Results                            
Dep. Variable:         donation_dummy   R-squared:                       0.000
Model:                            OLS   Adj. R-squared:                 -0.000
Method:                 Least Squares   F-statistic:                    0.3567
Date:                Thu, 25 Jan 2024   Prob (F-statistic):              0.550
Time:                        22:44:57   Log-Likelihood:                 15783.
No. Observations:               29806   AIC:                        -3.156e+04
Df Residuals:                   29804   BIC:                        -3.155e+04
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
Intercept      0.0200      0.001     14.085      0.0

The intercept coefficient, 0.0200, indicates the response rate in the control group, on average, in blue states.

The treatment coefficient, 0.0010, indicates the variation in the response rate between the treatment group and the control group in blue states.

The treatment is more effective in red states due to the larger treatment coefficient in red states.

## Question 2.

In [27]:
red_coef = model_red.params['treatment']
red_inter = model_red.params['Intercept']
blue_coef = model_blue.params['treatment']
blue_inter = model_blue.params['Intercept']
diff = round((red_coef + red_inter) - (blue_coef + blue_inter), 
             6)
diff

0.002308

The treatment coefficients have causal interpretation in each of the two regressions, due to randomization in each group.

The difference in the treatment effect between states does not necessarily have a causal interpretation states are not randomly assigned in real world.

# 4 Response rates and donation amount

## Question 1.

In [32]:
import pandas as pd
import statsmodels.api as sm
import statsmodels.formula.api as smf

df = pd.read_csv("charitable_giving.csv")

model = smf.ols(formula = 'donation_amount ~ treatment',
               data = df).fit()
print(model.summary())

                            OLS Regression Results                            
Dep. Variable:        donation_amount   R-squared:                       0.000
Model:                            OLS   Adj. R-squared:                  0.000
Method:                 Least Squares   F-statistic:                     3.461
Date:                Thu, 25 Jan 2024   Prob (F-statistic):             0.0628
Time:                        23:10:06   Log-Likelihood:            -1.7946e+05
No. Observations:               50083   AIC:                         3.589e+05
Df Residuals:                   50081   BIC:                         3.589e+05
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
Intercept      0.8133      0.067     12.063      0.0

The intercept coefficient, 0.8133, indicates the donors in the control group donates 0.8133 dollars, on average.
The treatment coefficient, 0.1536, indicates that donors in the treatement group donates 0.1536 dollars more than the control group, on average.


This difference has a causal interpretation because the regression is based on the full sample where treatment is randomly assigned.

## Question 2.

In [33]:
import pandas as pd
import statsmodels.api as sm
import statsmodels.formula.api as smf

df = pd.read_csv("charitable_giving.csv")
df_donate = df[df['donation_dummy'] == 1]

model = smf.ols(formula = 'donation_amount ~ treatment',
               data = df_donate).fit()
print(model.summary())

                            OLS Regression Results                            
Dep. Variable:        donation_amount   R-squared:                       0.000
Model:                            OLS   Adj. R-squared:                 -0.001
Method:                 Least Squares   F-statistic:                    0.3374
Date:                Thu, 25 Jan 2024   Prob (F-statistic):              0.561
Time:                        23:11:11   Log-Likelihood:                -5326.8
No. Observations:                1034   AIC:                         1.066e+04
Df Residuals:                    1032   BIC:                         1.067e+04
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
Intercept     45.5403      2.423     18.792      0.0

The intercept coefficient, 45.5403, indicates the donors in the control group donates 45.5403 dollars, on average.

The treatment coefficient, -1.6684, indicates that donors in the treatement group donates 1.67 dollars less than the control group, on average.

The treatment coefficient does not have a causal interpretation because treatment is not randomly assigned but decided by whether to donate.